sonar-flex-plugin/src/main/java/org/sonar/plugins/flex/FlexTokensVisitor.java

```diff
@@ -0,0 +1,117 @@
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.flex;
+
+import com.google.common.collect.Sets;
+import com.sonar.sslr.api.AstNode;
+import com.sonar.sslr.api.GenericTokenType;
+import com.sonar.sslr.api.Token;
+import com.sonar.sslr.api.TokenType;
+import com.sonar.sslr.api.Trivia;
+import com.sonar.sslr.impl.Lexer;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.cpd.NewCpdTokens;
+import org.sonar.api.batch.sensor.highlighting.NewHighlighting;
+import org.sonar.api.batch.sensor.highlighting.TypeOfText;
+import org.sonar.flex.api.FlexKeyword;
+import org.sonar.flex.api.FlexPunctuator;
+import org.sonar.flex.api.FlexTokenType;
+import org.sonar.squidbridge.SquidAstVisitor;
+import org.sonar.sslr.parser.LexerlessGrammar;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.Set;
+
+public class FlexTokensVisitor extends SquidAstVisitor<LexerlessGrammar> {
+
+  private static final String NORMALIZED_CHARACTER_LITERAL = "$CHARS";
+  private static final String NORMALIZED_NUMERIC_LITERAL = "$NUMBER";
+  private static final Set<FlexKeyword> KEYWORDS = Sets.immutableEnumSet(Arrays.asList(FlexKeyword.values()));
+
+  private final SensorContext context;
+  private final Lexer lexer;
+
+  public FlexTokensVisitor(SensorContext context, Lexer lexer) {
+    this.context = context;
+    this.lexer = lexer;
+  }
+
+  @Override
+  public void visitFile(@Nullable AstNode astNode) {
+    NewHighlighting highlighting = context.newHighlighting();
+    File file = getContext().getFile();
+    InputFile inputFile = context.fileSystem().inputFile(context.fileSystem().predicates().is(file));
+    highlighting.onFile(inputFile);
+
+    NewCpdTokens cpdTokens = context.newCpdTokens();
+    cpdTokens.onFile(inputFile);
+
+    Iterator<Token> iterator = lexer.lex(file).iterator();
+    // we currently use this hack to remove "import" directives
+    boolean importDirective = false;
+    while (iterator.hasNext()) {
+      Token token = iterator.next();
+      TokenType tokenType = token.getType();
+      if (tokenType == FlexKeyword.IMPORT) {
+        importDirective = true;
+      } else if (importDirective) {
+        // We do nothing as we want to ignore "import" directives
+        if (tokenType == FlexPunctuator.SEMI) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)